File: ios/EnterpriseMobileApp/Sources/Core/Domain/Interfaces/CrashReporter.swift
import Foundation

/// Observability interface for crash reporting - no SDK bleed into Domain/Presentation.
public protocol CrashReporter: Sendable {
    func logNonFatal(_ error: Error)
    func setCustomKey(_ key: String, value: String)
    func logBreadcrumb(_ message: String)
}

File: android/core/domain/src/main/java/com/enterprise/core/domain/interfaces/CrashReporter.kt
package com.enterprise.core.domain.interfaces

/** Crash reporting interface - implemented in Infrastructure layer. No SDK bleed. */
interface CrashReporter {
    fun logNonFatal(exception: Exception)
    fun setCustomKey(key: String, value: String)
    fun logBreadcrumb(message: String)
}
